src/main/java/com/google/devtools/build/lib/analysis/starlark/StarlarkTransition.java

```diff
@@ -381,6 +338,74 @@ private static Target getActual(
     return cleanedOptionMap.buildOrThrow();
   }
 
+  /**
+   * Validate the value of a particular build setting after a transition has been applied.
+   * @param buildSettingRule the build setting to validate.
+   * @param options the {@link BuildOptions} reflecting the post-transition configuration.
+   * @param maybeAliasSetting the label used to refer to the build setting in the transition,
+   *                          possibly an alias. This is only used for error messages.
+   * @param inputAndOutputSettings the transition input and output settings. This is only used for
+   *                               error messages.
+   * @return {@code true} if and only if the setting is set to its default value after the
+   *         transition.
+   * @throws TransitionException if the value returned by the transition for this setting has an
+   *                             invalid type.
+   */
+  private static boolean validateAndCheckIfAtDefault(Rule buildSettingRule, BuildOptions options,
+      Label maybeAliasSetting, Set<Label> inputAndOutputSettings) throws TransitionException {
+    // If the build setting was *not* referenced in the transition by an alias, this is the same
+    // value as {@code maybeAliasSetting}.
+    Label actualSetting = buildSettingRule.getLabel();
+    Object newValue = options.getStarlarkOptions().get(actualSetting);
+    // TODO(b/154132845): fix NPE occasionally observed here.
+    Preconditions.checkState(
+        newValue != null,
+        "Error while attempting to validate new values from starlark"
```

Comment:
@gregestren Do you know whether this is still needed? It is the only use of the parameter `inputAndOutputSettings`.

Reply:
Good question. The bug says there were instances of NPes but we haven't figured out how to reproduce them. So this at least provides a clear error message vs. an arbitrary crash.

Given that I'd keep this, even with the extra complication it adds.